### PR TITLE
Add usage example to APIDoc and updated README

### PR DESCRIPTION
**Pull request by helenmasters**
* Added usage example to APIDoc for Session and CookieParameters
 * Added further descriptions to parameters in APIDoc
 * Fixed typos in APIDoc
 * Added APIDoc link to the README header
 * Added missing standard sections to README
 * Reworded parts of the README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,56 +21,79 @@
...
+import Kitura
 import KituraSession
 import KituraSessionRedis
 
 let redisStore = RedisStore(redisHost: host, redisPort: port)
 let session = Session(secret: "Some secret", store: redisStore)
 router.all(middleware: session)
 ```
-First an instance of `RedisStore` is created (see [`KituraSessionRedis`](https://github.com/IBM-Swift/Kitura-Session-Redis) for more information), then an instance of `Session` with the store as parameter is created, and finally it is connected to the desired path.
+First an instance of `RedisStore` is created (see [`KituraSessionRedis`](https://github.com/IBM-Swift/Kitura-Session-Redis) for more information), then an instance of `Session` with the store as a parameter is created, then it is connected to the desired path.
```
Comment (djones6) [inline README.md]: I'd prefer
> An instance of `RedisStore` is created that will be used to persist session data (see [`KituraSessionRedis`](https://github.com/IBM-Swift/Kitura-Session-Redis) for more information).  An instance of `Session` is then created, specifying *redisStore* as the session store.  Finally, the *session* instance is registered as a middleware on the desired path.

File: Sources/KituraSession/CookieParameter.swift [hunk 2]
```diff
@@ -18,18 +18,28 @@ import Foundation
...
+/// ```
+/// In the example, an instance of `Session` is created with a custom value for the `CookieParameter` name.
 public enum CookieParameter {
 
-    /// The cookie's name.
+    /// The cookie's name, defaults to "kitura-session-id".
     case name(String)
 
-    /// The cookie's Path attribute.
+    /// The cookie's path attribute, this defines the path for which this cookie should be supplied.
```
Comment (djones6) [inline Sources/KituraSession/CookieParameter.swift]: > The cookie's path attribute. This specifies the path for which the cookie is valid. The client should only provide this cookie for requests on this path.

File: Sources/KituraSession/CookieParameter.swift [hunk 3]
```diff
@@ -18,18 +18,28 @@ import Foundation
...
+/// ### Usage Example: ###
+/// ```swift
+/// let session = Session(secret: "Something very secret", cookie: [.name("mySessionId")])
+/// router.all(middleware: session)
+/// ```
+/// In the example, an instance of `Session` is created with a custom value for the `CookieParameter` name.
 public enum CookieParameter {
 
-    /// The cookie's name.
+    /// The cookie's name, defaults to "kitura-session-id".
```
Comment (djones6) [inline Sources/KituraSession/CookieParameter.swift]: name, defaults to  ->  name. Defaults to

File: Sources/KituraSession/CookieParameter.swift [hunk 4]
```diff
@@ -18,18 +18,28 @@ import Foundation
...
     case name(String)
 
-    /// The cookie's Path attribute.
+    /// The cookie's path attribute, this defines the path for which this cookie should be supplied.
+    /// Defaults to "/" which means allow any path.
     case path(String)
 
-    /// The cookie's Secure attribute.
+    /// The cookie's secure attribute, this indicates whether the cookie should be provided only
```
Comment (djones6) [inline Sources/KituraSession/CookieParameter.swift]: attribute, this indicates whether  ->  attribute, indicating whether

**Review by djones6 — CHANGES_REQUESTED** [review]
Good fixes, I feel I'm nitpicking here but some wording suggestions.

File: Sources/KituraSession/Session.swift [hunk 5]
```diff
@@ -22,6 +22,15 @@ import Foundation
...
 /// A pluggable middleware for managing user sessions.
+///
+/// In order to use the Session middleware, an instance of `Session` has to be created. In the example
+/// below an instance of `Session` is created, then it is connected to the desired path.
+///
+/// ### Usage Example: ###
+/// ```swift
+/// let session = Session(secret: "Something very secret")
+/// router.all(middleware: session)
+/// ```
```
Comment (Andrew-Lees11) [inline Sources/KituraSession/Session.swift:45]: Can we add an example of using the session within a route? E.g.
```
router.post("/session") {request, response, next in
            request.session?["key"] = "value"
            next()
}
router.get("/session") {request, response, next in
            let value = request.session?["key"] as? String
            next()
}
```

File: README.md [hunk 6]
```diff
@@ -21,56 +21,79 @@
...
 
 ```swift
+import Kitura
 import KituraSession
 import KituraSessionRedis
 
 let redisStore = RedisStore(redisHost: host, redisPort: port)
 let session = Session(secret: "Some secret", store: redisStore)
 router.all(middleware: session)
 ```
```
Comment (Andrew-Lees11) [inline README.md:85]: Could we add using the session within a route to this example. E.g:
```
 router.post("/session") {request, response, next in
            request.session?["key"] = "value"
            next()
}
router.get("/session") {request, response, next in
            let value = request.session?["key"] as? String
            next()
}
```

**Comment by Andrew-Lees11** [comment]
I have addressed mine and @djones6 review comments and rebased this PR on top of the current master. 
I also added readme examples for Codable and TypeSafeSession to the README.

**Review by Andrew-Lees11 — APPROVED** [review]

**Review by djones6 — APPROVED** [review]

**Pull request merged**